python/spark_sklearn/grid_search.py

```diff
@@ -277,7 +278,11 @@ def _fit(self, X, y, groups, parameter_iterable):
 
         estimator = self.estimator
         cv = check_cv(self.cv, y, classifier=is_classifier(estimator))
-
+        
+        if hasattr(cv, 'random_state'):
```

Comment:
why is this necessary?

Reply:
Not all of the CV split implementations have a random element to them

Reply:
Since, "An iterable yielding train/test splits" is a legal argument to `check_cv`. I believe one might be able to write an implementation of cv without a `random_state`.  An example could be [LeaveOneOut](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.LeaveOneOut.html).